docs/question.md

```diff
@@ -447,3 +447,84 @@ For most [elements] there are four different ways of grading the student answer.
 4. Write an [external grader](externalGrading.md), though this is typically applied to more complex questions like coding.
 
 If a question has more than one of the above options, each of them overrides the one before it. Even if options 3 (custom grade function) or 4 (external grader) are used, then it can still be helpful to set a correct answer so that it is shown to students as a sample of what would be accepted. If there are multiple correct answers then it's probably a good idea to add a note with [`pl-answer-panel`](elements/#pl-answer-panel-element) that any correct answer would be accepted and this displayed answer is only an example.
+
+## Initializing parameters from outside the question
+
+The `data["params"]` dictionary normally used to pass parameters to `question.html` can be &quot;seeded&quot; from outside the question&apos;s files, allowing some portion of its randomization to be set by an assessment, course instance, or course configuration file. Such "`questionParams`" can change the behavior of a question without needing to alter its source code, so a question can be reused on different assessments but with a different configuration of parameters to alter its actual variants as needed.
+
+A `questionParams` is a JSON object that can be placed in three different PrairieLearn file types. When a question&apos;s `generate()` function is called, the `data["params"]` dictionary will be pre-filled with the key-value pairs defined by `questionParams` in these locations:
+
+1. `infoCourse.json`: parameters defined here will be passed to all questions in the course, including when they are previewed by an instructor without being attached to an assessment.
+2. `infoCourseInstance.json`: these parameters will be passed to questions when they are used in one of this course instance's assessments.
+3. `infoAssessment.json`: parameters can be added to the assessment itself, to one of its zones, to one of its lists of alternatives, or to an individual question object. In each case, the parameters will be passed to all questions nested under that element. For example, parameters specified at the top-level assessment apply to every question; paramters for a zone only apply to the questions in that zone; and parameters on an individual question only apply to that question.
+
+Parameters with matching names are overridden by the locations that are farther down the preceding list. (A parameter configured by `infoCourse.json` will be supplanted by one configured in `infoCourseInstance.json`, parameters on an individual question in an assessment will override all others.)
```

Comment:
It might be worth calling out explicitly that these are not deep merged - that is, if I have `{"a": { "b": 1 } }` at the course level and `{"a": { "c": 2 } }` at the question level, you end up with `{"a": { "c": 2 } }`, not `{"a": { "b": 1, "c": 2 } }`.